main.py

```diff
@@ -81,16 +79,31 @@
     # perform splitting
     X_train, y_train, X_test, y_test = ttsplit.fSplitDataset(dAllPatches, dAllLabels, dAllPats, cfg['sSplitting'], patchSize, cfg['patchOverlap'], cfg['dSplitval'], '')
 
+    if patchSize[0] != 40:
+        X_train, X_test, NewpatchSize = scaling.fscalling(X_train, X_test, patchSize)
+        sDatafile = sOutPath + os.sep + sFSname + ''.join(map(str, patchSize)).replace(" ", "") +'to'+''.join(map(str, NewpatchSize)).replace(" ", "") + '.h5'
+        OrigPatchSize = patchSize
+        patchSize = NewpatchSize
+     
     # save to file (deprecated)
     # sio.savemat(sOutPath + os.sep + sFSname + str(patchSize[0]) + str(patchSize[1]) + '_input.mat', {'X_train': X_train, 'y_train': y_train, 'X_test': X_test, 'y_test': y_test, 'patchSize': cfg['patchSize']})
     with h5py.File(sDatafile, 'w') as hf:
-        hf.create_dataset('X_train', data=X_train)
-        hf.create_dataset('X_test', data=X_test)
-        hf.create_dataset('y_train', data=y_train)
-        hf.create_dataset('y_test', data=y_test)
-        hf.create_dataset('patchSize', data=patchSize)
-        hf.create_dataset('patchOverlap', data=cfg['patchOverlap'])
+         hf.create_dataset('X_train', data=X_train)
+         hf.create_dataset('X_test', data=X_test)
+         hf.create_dataset('y_train', data=y_train)
+         hf.create_dataset('y_test', data=y_test)
+         hf.create_dataset('patchSize', data=patchSize)
+         hf.create_dataset('patchOverlap', data=cfg['patchOverlap'])
+         hf.create_dataset('OrigPatchSize', data=OrigPatchSize)
```

Comment:
How does patchsize look like resp. where do you define the scaled patchsizes?
I would suggest to use the config patchSize parameter as follows:
patchSize = [x1, y1, z1, x2, y2, z2, ...]
where x1 corresponds to 1st patching along x direction, y1: 1st patching along y direction z1: 1st patching along z direction, so e.g. for the 2D case with 2 patchSizes: patchSize = [40, 40, , 20, 20, ]
=> makes OrigPatchSize obsolete

Reply:
The OrigiPatchSize was designed to creat the saving output path. e.g. the training results when using 4040 patches will be saved in path '~/headcross/4040/testout'. If the pateches are resized from 8080 to 4040, the result should be saved in '~/headcross/8080/testout'.